llvm/lib/CodeGen/ScheduleDAGInstrs.cpp

```diff
@@ -1113,7 +1113,7 @@ static void toggleKills(const MachineRegisterInfo &MRI, LivePhysRegs &LiveRegs,
       continue;
 
     // Things that are available after the instruction are killed by it.
-    bool IsKill = LiveRegs.available(MRI, Reg);
+    bool IsKill = LiveRegs.available(Reg);
```

Comment:
Shouldn't set kill on reserved registers, although you might need an additional check. I think it would be best if the verifier didn't allow it

Reply:
What do you mean by "the verifier" since I do not see any verifier in this code

Reply:
It means that `lib/CodeGen/MachineVerifier` could disallow killing a reserved register.